src/main/java/spoon/support/reflect/reference/CtReferenceImpl.java

```diff
@@ -101,9 +105,9 @@ private void checkIdentiferForJLSCorrectness(String simplename) {
 		 */
 		//JDTTreeBuilderHelper.computeAnonymousName returns "$numbers$Name" so we have to skip them if they start with numbers
 		//allow empty identifier because they are sometimes used.
-		if (!simplename.matches("<.*>|\\d.*|^.{0}$")) {
+		if (!IS_INNER_OR_GENERIC_OR_EMPTY.matcher(simplename).matches()) {
```

Comment:
The more I look at this, the more I'm thinking it's completely unnecessary to use a regex for it. The regex only actually checks the first and last characters so it seems redundant to use a regex at all. Seems to me like something like this should be faster and work just as well:

```java
	private static boolean isEmptyOrLocalOrGeneric(String identifier) {
		return identifier.isEmpty()
				|| Character.isDigit(identifier.charAt(0))
				|| (identifier.startsWith("<") && identifier.endsWith(">"));
	}
```

WDYT?

Reply:
oh nice, a human, static compilation of the regexp :) :+1:

Reply:
This was a misunderstanding on my side, it's actually not about generics but about names like `<init>` there. I re-used your code there, the check for anonymous/local classes was moved to the other part of the code.